test/unit/cursor/run_command_cursor.test.ts

```diff
@@ -0,0 +1,91 @@
+import { expect } from 'chai';
+
+import { MongoAPIError, MongoClient, RunCommandCursor } from '../../mongodb';
+
+describe('class RunCommandCursor', () => {
+  let client: MongoClient;
+  beforeEach(async function () {
+    client = new MongoClient('mongodb://iLoveJavascript');
+  });
+
+  context('constructor()', () => {
+    it('freezes and stores the command on the cursor instance', () => {
+      const cursor = client.db().runCursorCommand({ a: 1 });
+      expect(cursor).to.have.property('command').that.is.frozen;
+    });
+
+    it('creates getMoreOptions property with no defaults', () => {
+      const cursor = client.db().runCursorCommand({ a: 1 });
+      expect(cursor).to.have.property('getMoreOptions').that.deep.equals({});
+    });
+  });
+
+  context('setComment()', () => {
+    it('stores the comment value in getMoreOptions', () => {
+      const cursor = client.db().runCursorCommand({ a: 1 });
+      cursor.setComment('iLoveJS');
+      expect(cursor).to.have.nested.property('getMoreOptions.comment', 'iLoveJS');
+    });
+  });
+
+  context('setMaxTimeMS()', () => {
+    it('stores the maxTimeMS value in getMoreOptions.maxAwaitTimeMS', () => {
+      const cursor = client.db().runCursorCommand({ a: 1 });
+      cursor.setMaxTimeMS(2);
+      expect(cursor).to.have.nested.property('getMoreOptions.maxAwaitTimeMS', 2);
+    });
+
+    it('does not throw on incorrect maxTimeMS type', () => {
+      const cursor = client.db().runCursorCommand({ a: 1 });
+      // @ts-expect-error: testing for incorrect type
+      cursor.setMaxTimeMS('abc');
+      expect(cursor).to.have.nested.property('getMoreOptions.maxAwaitTimeMS', 'abc');
+    });
+  });
+
+  context('setBatchSize()', () => {
+    it('stores the batchSize value in getMoreOptions', () => {
+      const cursor = client.db().runCursorCommand({ a: 1 });
+      cursor.setBatchSize(2);
+      expect(cursor).to.have.nested.property('getMoreOptions.batchSize', 2);
+    });
+
+    it('does not throw on incorrect batchSize type', () => {
+      const cursor = client.db().runCursorCommand({ a: 1 });
+      // @ts-expect-error: testing for incorrect type
+      cursor.setBatchSize('abc');
+      expect(cursor).to.have.nested.property('getMoreOptions.batchSize', 'abc');
+    });
+  });
+
+  context('clone()', () => {
+    it('returns a new RunCursorCommand instance', () => {
```

Comment:
should there be some more tests here about the effect of cloning a cursor with set comment, maxTimeMS, and batchSize?

Reply:
Good call, currently the getMore options would be lost in the clone. I had considered making the clone API unsupported for this cursor type, the API is not super valuable (you can put your db.runCursorCommand API into a recallable function), and has these edge cases we need to account for. Thoughts?

Reply:
is clone spec for all cursors? if not, I don't see why we need to implement it if no one is asking for it

Reply:
Not spec, sounds good to me